piptools/resolver.py

```diff
@@ -416,6 +423,8 @@ def _iter_dependencies(
         self, ireq: InstallRequirement
     ) -> Iterator[InstallRequirement]:
         """
+        Collect all secondary dependencies for an ireq.
```

Comment:
I usually go for "emit" in generator functions' docstrings. I feel like this word represents better that it can produce/stream more data, than just a single `return` instruction does.
```suggestion
        Emit all secondary dependencies for an ireq.
```